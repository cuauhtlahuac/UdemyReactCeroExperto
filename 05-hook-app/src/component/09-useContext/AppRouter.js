import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import NavBar from './NavBar';

import AboutScreen from './AboutScreen';
import LoginScreen from './LoginScreen';
import HomeScreen from './HomeScreen';
import NotFound from './NotFound';

const AppRouter = () => {
  return (
    <div>
      <Router>
        <NavBar />

        <Switch>

        <Route exact path="/about" component={ AboutScreen } />
        <Route exact path="/login" component={ LoginScreen } />
        <Route exact path="/" component={ HomeScreen } />
        <Route component={NotFound} />
            
        </Switch>
      </Router>
    </div>
  )
}

export default AppRouter
