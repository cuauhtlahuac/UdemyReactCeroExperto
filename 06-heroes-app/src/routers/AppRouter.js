import React, { useContext } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import { AuthContext } from 'contexts/authContext';
import LoginScreen from 'components/login/LoginScreen';

import DashboardRoutes from './DashboardRoutes';
import PrivateRoute from './PrivateRoute';
import PublicRoutes from './PublicRoutes';


const AppRouter = () => {

	const { user } = useContext(AuthContext)
	
	return (
		<Router>
			<div>
				<Switch>
					<PublicRoutes
						path="/login" 
						component={ LoginScreen } 
						exact
					/>

					<PrivateRoute 
						path="/" 
						component={ DashboardRoutes }
						isAuthenticated={ user.logged }
					/>
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
