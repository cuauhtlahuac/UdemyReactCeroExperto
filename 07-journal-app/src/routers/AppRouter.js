import React, { useEffect, useState } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import { firebase } from 'firebase/firebase-config';

import { useDispatch } from 'react-redux';
import { loginAction } from 'actions/auth';
import { startLoadingNotesAction } from 'actions/notes';
import JournalScreen from 'components/journal/JournalScreen';

import PublicRoutes from './PublicRoutes';
import PrivateRoutes from './PrivateRoutes';
import AuthRouter from './AuthRouter';

const AppRouter = props => {
	const dispatch = useDispatch();

	const [ checking, setChecking ] = useState(true);

	const [ isLogged, setIsLogged ] = useState(false);

	useEffect(
		() => {
			firebase.auth().onAuthStateChanged(async user => {
				// es un observable
				if (user && user.uid) {
					// solo usuarios autenticados
					dispatch(loginAction(user.uid, user.displayName));

					dispatch(startLoadingNotesAction(user.uid));

					setIsLogged(true);
				} else {
					setIsLogged(false);
				}

				setChecking(false);
			});
		},
		[ dispatch, setChecking, setIsLogged ],
	);

	if (checking)
		return (
			<div className="general-center">
				<img
					src="https://www.downgraf.com/media/2019/05/Loader-animation-principle-freebie.gif"
					alt="Loader Animation"
				/>
			</div>
		);

	return (
		<Router>
			<Switch>
				<PublicRoutes
					path="/auth"
					component={AuthRouter}
					isAuthenticated={isLogged}
				/>

				<PrivateRoutes
					path="/"
					component={JournalScreen}
					isAuthenticated={isLogged}
				/>
			</Switch>
		</Router>
	);
};

export default AppRouter;
