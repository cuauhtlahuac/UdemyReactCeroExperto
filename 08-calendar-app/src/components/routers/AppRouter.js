import { useSelector } from 'react-redux';

import {
	BrowserRouter as Router,
	Switch,
	Redirect,
} from 'react-router-dom';

import LoginScreen from 'components/auth/LoginScreen';
import CalendarScreen from 'components/calendar/CalendarScreen';

import PublicRoutes from './PublicRoutes';
import PrivateRoutes from './PrivateRoutes';

const AppRouter = () => {
	const { checked, checking } = useSelector(state => state.auth);

	if (checking) {
		return <h5>Loading...</h5>;
	}

	return (
		<Router>
			<div>
				<Switch>
					<PrivateRoutes
						path="/"
						component={CalendarScreen}
						isAuthenticated={checked}
						exact
					/>

					<PublicRoutes
						path="/login"
						component={LoginScreen}
						isAuthenticated={checked}
						exact
					/>

					<Redirect to="/" />
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
