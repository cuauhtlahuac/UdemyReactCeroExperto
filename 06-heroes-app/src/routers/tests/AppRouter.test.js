import { AuthContext } from 'contexts/authContext';
import { shallow, mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import AppRouter from 'routers/AppRouter';

describe('<PrivateRoute /> Tests', () => {
	let component;
	const contextValue = {
		dispatch: jest.fn(),
		user: { logged: false },
	};

	beforeEach(() => {
		component = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>,
		);
	});

	test('should show component if is not Authenticated and save in localStorage', () => {
		expect(component).toMatchSnapshot();
	});
});
