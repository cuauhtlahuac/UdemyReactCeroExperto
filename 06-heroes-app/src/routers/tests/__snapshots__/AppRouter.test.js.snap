// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> Tests should show component if is not Authenticated and save in localStorage 1`] = `
<div>
  <div
    className="m-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
